torch/fx/passes/splitter_base.py

```diff
@@ -120,6 +119,16 @@ def __init__(
             "we might not care about non-tensor data flow and we can set this option "
             "to true to disable the functionality that prevent non-tensor data flow.",
         )
+        parser.add_argument(
+            "--move-non-tensor-nodes-on-boundary",
+            "--move_non_tensor_nodes_on_boundary",
+            required=False,
+            type=bool,
```

Comment:
From [argparse docs](https://docs.python.org/3/library/argparse.html#:~:text=The%20bool()%20function%20is%20not%20recommended%20as%20a%20type%20converter.%20All%20it%20does%20is%20convert%20empty%20strings%20to%20False%20and%20non%2Dempty%20strings%20to%20True.%20This%20is%20usually%20not%20what%20is%20desired.)

> The [bool()](https://docs.python.org/3/library/functions.html#bool) function is not recommended as a type converter. All it does is convert empty strings to False and non-empty strings to True. This is usually not what is desired.

```suggestion
            action='store_true',
```